system/Throttle/Throttler.php

```diff
@@ -139,7 +139,7 @@ public function check(string $key, int $capacity, int $seconds, int $cost = 1):
         // How many seconds till a new token is available.
         // We must have a minimum wait of 1 second for a new token.
         // Primarily stored to allow devs to report back to users.
-        $newTokenAvailable = (int) ($refresh - $elapsed - $refresh * $tokens);
+        $newTokenAvailable = (int) round((1 - $tokens) * $refresh);
```

Comment:
Why is `$elapsed` gone?

Reply:
Because `$tokens` at this moment is already `$old_token + $rate * $elapsed`, in other words, the new number of tokens after `$elapsed` seconds. The old formula indirectly substracted `$refresh` by elapsed twice, making it look like the countdown is faster than it should actually be.

The unit tests I commited proofed my assumption. Every "actual" tokenTime was much lower than the "expected" tokenTIme.

Reply:
Okay, thanks.